plugin/src/main/java/org/opensearch/ml/processor/MLInferenceSearchResponseProcessor.java

```diff
@@ -155,6 +154,8 @@ public void processResponseAsync(
         try {
             SearchHit[] hits = response.getHits().getHits();
             // skip processing when there is no hit
+
+            String queryString = request.source().toString();
```

Comment:
Does this also have `size` `sort` , or just contain `query` part ?

Reply:
added test testProcessResponseSuccessReadRequestMetaFieldFromInputMap